airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -265,8 +266,11 @@ private void reportFailure(final ConnectionUpdaterInput connectionUpdaterInput,
 
       if (autoDisableConnectionVersion != Workflow.DEFAULT_VERSION) {
         final AutoDisableConnectionActivityInput autoDisableConnectionActivityInput =
-            new AutoDisableConnectionActivityInput(connectionId, Instant.ofEpochMilli(Workflow.currentTimeMillis()));
-        runMandatoryActivity(autoDisableConnectionActivity::autoDisableFailingConnection, autoDisableConnectionActivityInput);
+            new AutoDisableConnectionActivityInput(connectionId, Instant.ofEpochMilli(Workflow.currentTimeMillis()),
+                workflowInternalState.getLastAutoDisableWarningTimestamp());
```

Comment:
Is it working? I don't think this properly pass from 1 wf to another (to be check).

Reply:
From what I can tell, `workflowInternalState` doesn't get reset between runs, we only clear particular fields of the `workflowInternalState`:

https://github.com/airbytehq/airbyte/blob/7bb4baec2546cbf3f679a1aef02db13baa6475d8/airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java#L367-L377

So I think the `lastAutoDisableWarningTimestamp` will carry forward from one run to the next. I'm not sure what happens if a temporal workflow is terminated and re-created though. We probably won't be able to do much about that. @benmoriceau what do you think?

Reply:
@terencecho @pmossman, I can show it to you. I think the main reason is that we don't pass this information in a workflow input.

Reply:
Had a chat with benoit. Basically when the workflow continues as new, it is re-instantiated with inputs form `connectionUpdaterInput`.

Meaning everything the `workflowInternalState` is re-instantiated and not passed in through the input. The workflow state is passed in, but we only want that passed in so we can use the event listener for testing purposes, which is why we have `workflowState.reset()`.

We could add the warning notification state into `ConnectionUpdaterInput`, but that's probably not what we want to do. I'll spend a little more time thinking of other ways to get around this. Obviously putting the state in a database is an option, but I'm also hoping to avoid that.